### DISCUSSION TITLE
Flake setup w/ caching?

### DISCUSSION THREAD

**Discussion by k-e-l-p** [Jovian-Experiments/Jovian-NixOS#457] [Q&A] [answered]
Currently writing my steam deck NixOS+flake config that uses these modules. I.. think it's working, but my deck is now compiling the custom kernel and mesa version and it is taking a really long time now. Anyone figured out how to get a binary cache for the module working?

**Comment by samueldr** [comment, marked as answer]
There is no official cache available. That's the expected behaviour at this moment.

Note that making a cache that works well would require keeping up with the `nixos-unstable` and evaluating at least as frequently as channels update. In turn, this would require *some* additional setup for CD (continuous delivery) in this repo.

**Reply by k-e-l-p** [reply]
Thanks for the quick reply, 

am I right in guessing that the long compile times are a non-issue if I import the module as-is advised in the documentation, via `fetchTarball`?

**Reply by samueldr** [reply]
Nope; whichever way this is “`imports`”-ed into your NixOS configuration, the packages are built against the NixOS configuration's packages.

Another way to phrase this, is that whichever way it is included in the configuration, as long as the same revision of Nixpkgs is used, the same outputs will be produced.

(The pinned Nixpkgs in the repo only serves for running builds “directly against” the repo. A known good arbitrary revision for the packages to be built against, and docs.)
